### PR TITLE
Guide updates for disputes.

### PR DESCRIPTION
**Pull request by eskimor**
Missing review remarks of dispute distribution guide PR and integration of dispute distribution into other subsystems.

### CODE REVIEW

File: roadmap/implementers-guide/src/node/disputes/dispute-participation.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Output:
   - [RuntimeApiMessage][RuntimeApiMessage]
   - [CandidateValidationMessage][CandidateValidationMessage]
   - [AvailabilityRecoveryMessage][AvailabilityRecoveryMessage]
+  - [`AvailabilityStoreMessage`][AvailabilityStoreMessage]
```
Comment (pepyakin) [inline roadmap/implementers-guide/src/node/disputes/dispute-participation.md:15]: even though I prefer ticks, I think it'd be better to be consistent here.
```suggestion
  - [AvailabilityStoreMessage][AvailabilityStoreMessage]
```

File: roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md [hunk 2]
```diff
@@ -59,6 +59,13 @@ struct State {
 }
 ```
 
+### On startup
+
+Check DB for recorded votes for non concluded disputes we have not yet
+recorded a local statement for.
+For all of those send `DisputeParticipationMessage::Participate` message to
```
Comment (rphmeier) [inline roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md:66]: 👍

File: roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md [hunk 3]
```diff
@@ -80,18 +87,39 @@ Do nothing.
...
+4. If candidate votes is empty and the statements only contain dispute-specific
+   votes, respond with `ImportStatementsResult::InvalidImport` and return.
+5. Otherwise, if there is already an entry from the validator in the respective
+  `valid` or `invalid` field of the `CandidateVotes`,  respond with
+  `ImportStatementsResult::ValidImport` and return.
+6. Add an entry to the respective `valid` or `invalid` list of the
+   `CandidateVotes` for each statement in `statements`.
+7. If the both `valid` and `invalid` lists now became non-zero length where
+   previously one or both had zero length, the candidate would be freshly
+   disputed now.
```
Comment (rphmeier) [inline roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md:105]:
```suggestion
7. If both the `valid` and `invalid` lists now have non-zero length where
   previously one or both had zero length, the candidate is now freshly disputed.
```

File: roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md [hunk 4]
```diff
@@ -80,18 +87,39 @@ Do nothing.
...
+8. If the candidate is not freshly disputed as determined by 7, continue with
+   10. If it is freshly disputed now, load `"active-dsiputes"` and add the
```
Comment (rphmeier) [inline roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md:107]:
```suggestion
   10. If it is freshly disputed now, load `"active-disputes"` and add the
```

File: roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md [hunk 5]
```diff
@@ -80,18 +87,39 @@ Do nothing.
...
+   Wait for response on the `report_availability` oneshot. If available, continue
+   with 10. If not send back `ImportStatementsResult::InvalidImport` and return.
+10. Write the `CandidateVotes` to the underyling DB.
+11. Send back `ImportStatementsResult::ValidImport`.
+12. If the dispute now has supermajority votes in the "valid" direction,
+    according to the `SessionInfo` of the dispute candidate's session, remove
+    from `"active-disputes"`.
+13. If the dispute now has supermajority votes in the "invalid" direction, there
+    is no need to do anything explicitly. The actual rollback will be handled
+    during the active leaves update by observing digests from the runtime.
```
Comment (rphmeier) [inline roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md:121]: https://github.com/paritytech/polkadot/pull/3348

This diff contains changes to 'active-disputes' so we make a distinction between "recent" (everything in DISPUTE_WINDOW, concluded or not) and "active" (subset of recent which is either not concluded or was concluded within some small amount of time)

Due to this, we actually never prune anything from `active-disputes` until it's purged by session age.

**Review by rphmeier — APPROVED** [review]
Changes look fine but we need to reconcile with #3348

**Comment by eskimor** [comment]
> Changes look fine but we need to reconcile with #3348

Let's get #3348 merged - I will resolve conflicts afterwards.

**Pull request merged**